src/bun.js/bindings/BunProcess.cpp

```diff
@@ -655,121 +732,171 @@ static void onDidChangeListeners(EventEmitter& eventEmitter, const Identifier& e
         signalNumberToNameMap->add(SIGINT, signalNames[1]);
         signalNumberToNameMap->add(SIGQUIT, signalNames[2]);
         signalNumberToNameMap->add(SIGILL, signalNames[3]);
+#ifdef SIGTRAP
         signalNumberToNameMap->add(SIGTRAP, signalNames[4]);
+#endif
         signalNumberToNameMap->add(SIGABRT, signalNames[5]);
+#ifdef SIGIOT
         signalNumberToNameMap->add(SIGIOT, signalNames[6]);
+#endif
+#ifdef SIGBUS
         signalNumberToNameMap->add(SIGBUS, signalNames[7]);
+#endif
         signalNumberToNameMap->add(SIGFPE, signalNames[8]);
         signalNumberToNameMap->add(SIGKILL, signalNames[9]);
+#ifdef SIGUSR1
         signalNumberToNameMap->add(SIGUSR1, signalNames[10]);
+#endif
         signalNumberToNameMap->add(SIGSEGV, signalNames[11]);
+#ifdef SIGUSR2
         signalNumberToNameMap->add(SIGUSR2, signalNames[12]);
+#endif
+#ifdef SIGPIPE
         signalNumberToNameMap->add(SIGPIPE, signalNames[13]);
+#endif
+#ifdef SIGALRM
         signalNumberToNameMap->add(SIGALRM, signalNames[14]);
+#endif
         signalNumberToNameMap->add(SIGTERM, signalNames[15]);
+#ifdef SIGCHLD
         signalNumberToNameMap->add(SIGCHLD, signalNames[16]);
+#endif
+#ifdef SIGCONT
         signalNumberToNameMap->add(SIGCONT, signalNames[17]);
+#endif
+#ifdef SIGSTOP
         signalNumberToNameMap->add(SIGSTOP, signalNames[18]);
+#endif
+#ifdef SIGTSTP
         signalNumberToNameMap->add(SIGTSTP, signalNames[19]);
+#endif
+#ifdef SIGTTIN
         signalNumberToNameMap->add(SIGTTIN, signalNames[20]);
+#endif
+#ifdef SIGTTOU
         signalNumberToNameMap->add(SIGTTOU, signalNames[21]);
+#endif
+#ifdef SIGURG
         signalNumberToNameMap->add(SIGURG, signalNames[22]);
+#endif
+#ifdef SIGXCPU
         signalNumberToNameMap->add(SIGXCPU, signalNames[23]);
+#endif
+#ifdef SIGXFSZ
         signalNumberToNameMap->add(SIGXFSZ, signalNames[24]);
+#endif
+#ifdef SIGVTALRM
         signalNumberToNameMap->add(SIGVTALRM, signalNames[25]);
+#endif
+#ifdef SIGPROF
         signalNumberToNameMap->add(SIGPROF, signalNames[26]);
+#endif
         signalNumberToNameMap->add(SIGWINCH, signalNames[27]);
+#ifdef SIGIO
         signalNumberToNameMap->add(SIGIO, signalNames[28]);
+#endif
 #ifdef SIGINFO
         signalNumberToNameMap->add(SIGINFO, signalNames[29]);
 #endif
+#ifdef SIGSYS
         signalNumberToNameMap->add(SIGSYS, signalNames[30]);
+#endif
     });
 
     if (!signalToContextIdsMap) {
-        signalToContextIdsMap = new HashMap<int, HashSet<uint32_t>>();
+        signalToContextIdsMap = new HashMap<int, SignalHandleValue>();
     }
 
     if (auto signalNumber = signalNameToNumberMap->get(eventName.string())) {
+#if !OS(WINDOWS)
         if (signalNumber != SIGKILL && signalNumber != SIGSTOP) {
+#else
+        if (signalNumber != SIGKILL) {
```

Comment:
```suggestion
        if (signalNumber != SIGKILL && signalNumber != SIGSTOP) {
```
`SIGSTOP` should be ignored, it's not possible to have a handler for it

Reply:
oh wait, this is for windows